src/backend/sgx/thread.rs

```diff
@@ -171,9 +190,9 @@ impl super::super::Thread for Thread {
         //
         // After the sallyport rework, we can test the sallyport itself and
         // remove this logic.
-        if self.cssa > 0 {
+        if self.cssa == 1 || self.cssa == 2 {
```

Comment:
Just a reminder to follow up with Roman and see if we can follow the above comment and remove this.

Reply:
Yes, we can parse the sallyport block to determine what requests are there, but why would removing this conditional be a desirable thing? How would you differentiate between the two different blocks?
Even if there'd only be a single block, it's well-defined that sallyport is only executed in CSSA 1 (and in this PR, 1 and 2), so why even bother parsing anything in CSSA 0 or 3 ?

By the way, does the block get reset to 0 on each CSSA state decrease? (and if not, don't we run into a risk of executing an already-executed block again in CSSA 3?)

From the context provided by @haraldh today at the status meeting, I think the implementation is correct and we should simply remove the comment.

Reply:
I think we should test, if the sallyport does contain s.th. useful before processing it in that CSSA.

Reply:
And the "reset" should happen inside the shim after processing the host return data in the sallyport crate.

Reply:
`for item in block` ... this iterator should not return anything, if the sallyport was not used/was reset.

Reply:
removed the comment, though and added a new issue https://github.com/enarx/enarx/issues/2400